File: Server/src/main/java/core/game/content/dialogue/DenulthDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the denulth dialogue plugin.
 * @author 'Vexia
 * @version 1.0
 */
@Initializable
public final class DenulthDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code DenulthDialogue} {@code Object}.
	 */
	public DenulthDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code DenulthDialogue} {@code Object}.
	 * @param player the player.
	 */
	public DenulthDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new DenulthDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Hello!");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Welcome back friend!");
			stage = 1;
			break;
		case 1:
			interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "How goes your fight with trolls?");
			stage = 2;
			break;
		case 2:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Very good! We are winning.");
			stage = 3;
			break;
		case 3:
			interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Good luck!");
			stage = 4;
			break;
		case 4:
			end();
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 1060 };
	}

}

File: Server/src/main/kotlin/rs09/game/content/dialogue/region/burthorpe/DenulthDialogue.kt
package rs09.game.content.dialogue.region.burthorpe

import api.openDialogue
import core.game.content.dialogue.DialoguePlugin
import core.game.node.entity.npc.NPC
import core.game.node.entity.player.Player
import core.plugin.Initializable
import org.rs09.consts.NPCs
import rs09.game.content.quest.members.deathplateau.DenulthDialogueFile

/**
 * @author qmqz
 */

@Initializable
class DenulthDialogue(player: Player? = null) : DialoguePlugin(player){

    override fun open(vararg args: Any?): Boolean {
        npc = args[0] as NPC
        openDialogue(player, DenulthDialogueFile())
        return true
    }

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        return true
    }

    override fun newInstance(player: Player?): DialoguePlugin {
        return DenulthDialogue(player)
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.DENULTH_1060)
    }
}
